### PR TITLE
Fix: Use carousel behaviour from wcag

### PR DESCRIPTION
**Pull request by joe-allen-89**
Fixes: https://github.com/adaptlearning/adapt-contrib-hotgraphic/issues/243

### Fix
* Align the accessibility with wcag for carousel behaviour. 

[//]: # (List appropriate steps for testing if needed)
### Testing
1. Interact with the hotgraphic using a screen reader
2. Check that all items are read out as a list
3. Open a popup and ensure that nav buttons are reading out which item is next/back

### CODE REVIEW

**Comment by joe-allen-89** [comment]
Currently the popup numbers are visible to the screen reader, with updating the nav to read out the next/back item I think we should hide the numbering, what's other people's opinions on that?

File: templates/hotgraphic.hbs [hunk 1]
```diff
@@ -23,7 +23,7 @@
 
         <button class="btn-icon hotgraphic__pin item-{{@index}}{{#if _graphic._classes}} {{_graphic._classes}}{{/if}}{{#if _pin.src}} has-pin-image{{/if}}{{#if _isVisited}} is-visited{{/if}} js-hotgraphic-item-click" data-index="{{@index}}" style="top: {{{_top}}}%; left: {{{_left}}}%;">
 
-          <span class="aria-label">{{#if ../_useNumberedPins}}{{math @index "+" 1}} {{/if}}{{#if _pin.alt}}{{{compile _pin.alt}}}{{else}}{{{compile title}}}{{/if}}.</span>
+          <span class="aria-label">{{#if ../_useNumberedPins}}{{math @index "+" 1}} {{/if}}{{#if _pin.alt}}{{{compile _pin.alt}}}{{else}}{{{compile title}}}{{/if}}. Item {{math @index "+" 1}} of {{../_items.length}}</span>
```
[the remark this answers is not in the document]
Comment (StuartNicholls) [reply templates/hotgraphic.hbs:26]: The index between the forward and back arrows in the function updatePageCount should also be in plain English and translatable, rather than the format x / y

[the remark this answers is not in the document]
Comment (joe-allen-89) [reply templates/hotgraphic.hbs:26]: @StuartNicholls With the next and back arrows now reading out the item number and item I've suggested we hide the current x / y from the screen reader completely, do you think that would be better or prefer to have that spoken out in full English as well?

[the remark this answers is not in the document]
Comment (StuartNicholls) [reply templates/hotgraphic.hbs:26]: I was just referring to general legibility rather than screen readers.

I can't actually test the screen reader functionality at the moment but I'm not seeing that screen reader text on those buttons other than 'previous' and 'next' - just on the pins. But, if thats the case I'd imagine the page index wasn't necessary for screen readers.

[the remark this answers is not in the document]
Comment (StuartNicholls) [reply templates/hotgraphic.hbs:26]: Next / previous icons also need aria-hidden in the version I'm looking at

**Comment by StuartNicholls** [comment]
Looks good. 

One small thing that we might as well update here, we're in the process of moving the visited states to the beginning of the aria-label text for all instances (components/menus etc.) So..

hotgraphicView.js line 111 needs updating to accommodate this.

**Comment by joe-allen-89** [comment]
> @joe-allen-89 - Looks good.
> 
> One small thing that we might as well update here, we're in the process of moving the visited states to the beginning of the aria-label text for all instances (components/menus etc.) So..
> 
> hotgraphicView.js line 111 needs updating to accommodate this.
> 
> Additionally, we are updating all components with [aria-haspopup] see here:
> 
> [adaptlearning/adapt-contrib-core#245](https://github.com/adaptlearning/adapt-contrib-core/issues/245)
> 
> This will need adding to the hotgraphic pins

Excellent point, no time like the present. Both of those have been implemented.

**Review by StuartNicholls — APPROVED** [review]

**Review by chris-steele — APPROVED** [review]

**Review by cahirodoherty-learningpool — APPROVED** [review]
👀

**Comment by danielghost** [comment]
Requires `"inputType": "Text"` for the new schema additions.
